src/middlewared/middlewared/main.py

```diff
@@ -1855,15 +1856,31 @@ async def ws_handler(self, request):
                     )
                     break
 
-                if not connection.authenticated and len(msg.data) > 8192:
+                datalen = len(msg.data)
+
+                try:
+                    message = msg_size_check(connection.authenticated, datalen, msg.data)
```

Comment:
It parses JSON and then checks the message size. I think we can call it `parse_message` or something like this.

Reply:
OK. I'll rename.